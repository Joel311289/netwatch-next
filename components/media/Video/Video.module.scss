.wrapper {
  overflow: hidden;
  position: relative;
  background-color: var(--background-color-modal-video);

  &::before {
    content: '';
    display: block;
  }
}

.video {
  @include absolute(top 0 left 0 right 0 bottom 0);
  @include box(100%, 100%);

  border: 0;
}
