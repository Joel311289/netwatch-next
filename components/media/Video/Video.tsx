import { ElementHTML, Rectangle } from '@/types';
import { classes } from '@/utils/helpers';
import styles from '@/components/media/Video/Video.module.scss';
import { useSizeRatio } from '@/hooks/useSizeRatio';

export type VideoProps = typeof defaultProps &
  ElementHTML &
  Rectangle & {
    id: string;
    ratio?: number;
    autoplay?: boolean;
  };

const defaultProps = {
  ratio: 9 / 16
};

const Video = ({ className, id, autoplay, width, height, ratio }: VideoProps) => {
  const sizes = useSizeRatio({ width, height, ratio });

  return (
    <div className={classes(styles.wrapper)} aria-label="video">
      <iframe
        className={classes(styles.video, className)}
        src={id ? `https://www.youtube.com/embed/${id}?autoplay=${Number(autoplay)}` : ''}
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
        title="video"
      />

      <style jsx>{`
        .${styles.wrapper} {
          width: ${sizes.width || 'auto'};
          height: ${sizes.height || 'auto'};
          min-width: ${sizes.width || 'auto'};
        }
        .${styles.wrapper}::before {
          padding-top: ${ratio ? `${100 * ratio}%` : '0'};
        }
      `}</style>
    </div>
  );
};

Video.defaultProps = defaultProps;

export default Video;
