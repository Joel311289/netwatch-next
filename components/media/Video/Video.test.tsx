import { render, screen } from '@testing-library/react';
import Video from '@/components/media/Video/Video';

describe('Tests Video component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Video id="6Cl8PmVm3YE" />);

    expect(container).toMatchSnapshot();
  });

  test('renders with ratio correctly', () => {
    const width = 200;
    const ratio = 1.5;

    render(<Video id="6Cl8PmVm3YE" width={width} ratio={ratio} />);

    expect(screen.getByLabelText('video')).toHaveStyle(`width: ${width}px`);
    expect(screen.getByLabelText('video')).toHaveStyle(`height: ${width * ratio}px`);
  });
});
