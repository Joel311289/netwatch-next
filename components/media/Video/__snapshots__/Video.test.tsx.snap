// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Video component renders component correctly 1`] = `
<div>
  <div
    aria-label="video"
    class="wrapper"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      aria-label="video"
      class="video"
      src="https://www.youtube.com/embed/6Cl8PmVm3YE?autoplay=NaN"
      title="video"
    />
    <style>
      
        .wrapper {
          width: auto;
          height: auto;
          min-width: auto;
        }
        .wrapper::before {
          padding-top: 56.25%;
        }
      
    </style>
  </div>
</div>
`;
